idunn/utils/result_filter.py

```diff
@@ -221,21 +228,28 @@ def check(
         #   - over one of the result names
         #   - over one of the result names together with one of the admin names
         #   - over one of the result names together with a postcode
+        #   - postcode only, if the query consists of a single word
         def coverage(terms):
             return sum(
                 any(self.word_matches(query_word, term) for query_word in query_words)
                 for term in terms
             ) / len(terms)
 
-        check_coverage = any(
-            coverage(terms) >= (2 / 3)
+        candidate_terms_to_cover = (
+            terms
             for name_words in map(self.words, names)
             for terms in [
                 name_words,
                 *(name_words + self.words(admin) for admin in admins),
                 *(name_words + [postcode] for postcode in postcodes),
             ]
         )
+        if len(query_words) == 1:
+            candidate_terms_to_cover = chain(
+                candidate_terms_to_cover, ([postcode] for postcode in postcodes)
+            )
```

Comment:
I wonder if it would make it slightly more readable to use `candidate_terms_to_cover` as a list here? This is definitely opinionated and not important. I just, as usual, enjoy this kind of reflections ^^'

Reply:
In most cases, it's probably insignificant. But in some extreme cases, (with many names variants :heavy_multiplication_x:  many admins), there may be a real benefit to evaluate `candidate_terms_to_cover` lazily. So I tried to preserve this behavior.